### ISSUE TITLE
how to build and install in order to allow PlusLib on linux to see it

### ISSUE THREAD

**Issue by faustomilletari** [PlusToolkit/ndicapi#3]
Hello,
I'm trying to build PlusLib (actually PlusApp) on linux with NDI Polaris support.

I can do so on MacOS but not on linux. reason is ndicapi is not seen! 

i cloned this repository, and i have ran make with success. I have also switched on the option to have the dynamic lib compiled which gave me the .so file in addition to the .a file.

how can i tell to PlusLib where to find ndicapi? 

the error i encounter is an error message about the impossibly to find ndicapi from the CMake in PlusLib/src/PlusDataCollection/CMakeList.txt 

Thanks for your help.

**Comment by lassoan** [comment]
Probably you need a more recent version of CMake.

**Comment by faustomilletari** [comment]
I have 3.8.0 i think. Do i need to clean up my PlusApp build and restart from the beginning after i get ndicapi compiled? I also noticed that make install in ndicapi doesn't really do much apparently... How can i be sure that everything was installed in correct location and that CMake won't have problems finding it?

(sorry i'm not a great expert of CMake)

Thanks a lot for your answers so far :)

**Comment by lassoan** [comment]
Is this a build from scratch or you updated an older Plus build?

**Comment by faustomilletari** [comment]
I'm building from scratch, first time on ubuntu 16.04. the build failed because it was not finiding ndicapi. then i cloned this repo and did `cmake .` `make` and `make install`. this created a share folder in the directory tree with a `.a` file in it. Then i ran 'make' again in the PlusBuild directory, which unfortunately terminated with the same error about the fact that ndicapi was missing or not built

**Comment by lassoan** [comment]
Plus should and download build ndicapi. Do you see ndicapi source file in the deps directory (next to ITK and VTK)?

**Comment by faustomilletari** [comment]
seems that ndicapi is not there :( 
![buildplus](https://cloud.githubusercontent.com/assets/13331646/26514415/b0356d74-423f-11e7-9b12-68f33e9bbf07.png)

**Comment by faustomilletari** [comment]
ok. sorry. in ~/PlusBuild/build/Deps i can see ndicapi  and ndicapi-bin ndicapi-prefix

**Comment by faustomilletari** [comment]
is it maybe because i'm building into the directory build? (this anyway seems to happen only with ndicapi)

**Comment by lassoan** [comment]
OK, so Plus has downloaded and build ndicapi. Do you see ndicapi binary files in ~/PlusBuild/build/bin?

**Comment by faustomilletari** [comment]
i see a ndicapi.so there. makes sense!

**Comment by lassoan** [comment]
OK, so the problem is that PlusLib does not get ndicapi info. I check if I can reproduce the issue.

**Comment by faustomilletari** [comment]
one last thing. I was in master branch when this happened. Now i'm running another compile on Plus2.4 branch without using the build directory (compiling directly in PlusBuild).

Thanks a lot for your help. I'll update you if there are news

**Comment by lassoan** [comment]
Plus2.4 is very different. If you switch between master and 2.4 then you have to restart the build from scratch.

**Comment by faustomilletari** [comment]
yes. i'm doing it in another directory and i cloned it separately. actually for me it would be perfect if 2.4 works. With master branch i have also encountered other issues on macOs (i'm using 2.4 on macos). For example the program would crash when one would push the button during spatial calibration to set up the rough spatial sizes of the image pixels and therefore find the pattern. I can give a try again and maybe post here another issue if it's interesting to you to solve it on macos. 

On mac it's very simple to achieve a working system with all components in place. even ephipan framegrabbers can be easily installed once one has the library files which can be downloaded from ephipan website.

**Comment by faustomilletari** [comment]
I also find that plus would be sooo much popular if everything would be done with a website + a github repository. having everything here instead than on assembla would be so much better.

**Comment by lassoan** [comment]
When we started Plus, github barely existed and it was not suitable for proper project management. Github still does not have forum/messaging, it has no wiki, project/ticket management is still quite basic, there is no time tracking, customer support is arrogant/non-existent, and private project hosting is expensive, but it gets many things right and slowly catching up with implementing missing key features. We see (and enjoy) popularity of github, that's why we moved source code repositories here and did not disable issue submission. As github improves, we'll probably migrate more and more from Assembla.

**Comment by lassoan** [comment]
It's good to know that Epiphan works well on Mac (we implemented it so that it would, but probably never tested). If you find any issues with the master version, please report it, it would help us a lot.

**Comment by lassoan** [comment]
We've made a few small fixes and now latest master version builds fine for me on Linux. Let us know if you still have build problems.

**Comment by faustomilletari** [comment]
I have built on mac some time ago (Jan 2017 i think, commit 09fecad Jan 27th). I’m redoing it because i’m actually experiencing issues with both 2.4 and master on linux.

The only thing to do on mac in order to be able to build is ask brew to recompile qt5.7 with wxwidgets (or was is webwidgets) i think, which takes time, and then export the installation path of qt into CMAKE_PREFIX_PATH.

Ndi capture device (polaris) works as soon as one has the drivers installed.

For ephipan I remember I had to struggle a lot, but this was looooong ago. basically it was looking for .so and .a files from ephipan also on mac, while it should have been looking for .dylib

I will update you about compilation outcome on mac of both 2.4 and master. 

Actually I have been using mac for the past 10 years i think and therefore it’s possible that I am the reason why compilation does not succeed on linux.


Update 1) Compilation of master fails for the same ndicapi related issue on mac. 
Update 2) Compilation of Plus-2.4 on mac fails as well but i think that’s still a problem related to qt5 and therefore my fault.


Fausto
